libs/core-wrappers/src/lib/FieldWrapperFormik/FieldWrapperFormik.tsx

```diff
@@ -33,12 +33,12 @@ const FieldWrapper: React.FC<FieldWrapperProps> = ({
         ) : null}
       </Label>
       <Field name={name} as={Component} id={name} />
-      {meta.error && (
+      {meta.touched && meta.error && (
         <FormText className={styles['form-field__help']} color="body">
           {meta.error}
         </FormText>
       )}
-      {description && !meta.error && (
+      {description && (
```

Comment:
Please teach me what you learned and what this change offers.

I will work on this code soon, and want to benefit from your experience.

Reply:
My Formik knowledge comes from following the tutorial: [](https://formik.org/docs/tutorial).

`meta.touched` is Formik's way of checking if an input element''s `onBlur` event has been triggered -- it's a boolean, so true if it has, false otherwise. `meta.error` is Formik's way of providing the associated string error message for a given input element from the form validation. (These two variables (`meta.touched` and `meta.error`) are linked to a given input field and a Yup validation schema entry by providing a consistent `name` variable value to the Yup schema and components.)

What does this change offer (or why did I make it)?

 **When the form first renders the form validation is immediately run.** In our case, the fields we have marked as required in the Yup schema and have given empty strings as default values to will immediately fail (i.e. `meta.error` will be defined and truthy). With the pre-change code on L36, this results in `meta.error` strings being immediately displayed under such input fields when the form first renders, which isn't the way Core-Portal implements things -- onBlur ftw(?).

Looking at the original L36 and L41, you can see his original thought process: "If you have an error, then display it. If you have a description and you don't have an error, then display the description." I assume he wanted a "see-saw" for fields that had a description -- if you get an validation error, show the error but not the field description. At first I implemented what I thought he wanted, so I had `description && !(meta.touched && meta.error)` for L41. The behavior seemed a bit jarring to me and then I remembered one of Sophia's earlier PR's where we were okay with showing both the error and the description at the same time when there's an error. I also just checked Core-Portal's ticket create form and it shows the description and error at the same time when there's an error.

Reply:
Excellent. Thank you.